resources/providers/awslib/s3/provider.go

```diff
@@ -60,18 +61,28 @@ func (p Provider) DescribeBuckets(ctx context.Context) ([]awslib.AwsResource, er
 		// If the bucket is not in the configured region additional API calls for resources will fail, we should not
 		//  describe this bucket.
 		if bucketRegion != p.region {
-			log.Debugf("Bucket %s is in region %s and not in the configured region %s. Not describing this bucket", *clientBucket.Name, bucketRegion, p.region)
+			p.log.Debugf("Bucket %s is in region %s and not in the configured region %s. Not describing this bucket", *clientBucket.Name, bucketRegion, p.region)
 			continue
 		}
 
+		// Getting the bucket encryption, policy and versioning is not critical for the rest of the flow, so we should
+		//	keep describing the bucket even if getting these objects fails.
 		sseAlgorithm, encryptionErr := p.getBucketEncryptionAlgorithm(ctx, clientBucket.Name)
-		// Getting the bucket encryption is not critical for the rest of the flow, so we should keep describing the
-		//	bucket even if getting the bucket encryption fails.
 		if encryptionErr != nil {
 			p.log.Errorf("Could not get encryption for bucket %s. Error: %v", *clientBucket.Name, encryptionErr)
 		}
 
-		result = append(result, BucketDescription{*clientBucket.Name, sseAlgorithm})
+		bucketPolicy, policyErr := p.getBucketPolicy(ctx, clientBucket.Name)
```

Comment:
In case we got an error, does the relevant Rego code can handle nil `bucketPolicy`?

Reply:
Yes, you can see a test of it [here](https://github.com/elastic/csp-security-policies/blob/main/bundle/compliance/cis_aws/rules/cis_2_1_2/test.rego#L8)